token_update_transaction_hip540_e2e_test.go

```diff
@@ -0,0 +1,718 @@
+	require.NoError(t, err)
+
+	// Update supply key
+	tx, err := NewTokenUpdateTransaction().
+		SetTokenID(*receipt.TokenID).
+		SetSupplyKey(invalidSupplyKey).
+		SetKeyVerificationMode(NO_VALIDATION).
+		FreezeWith(env.Client)
+	assert.NoError(t, err)
+
+	_, err = supplyKey.SignTransaction(&tx.Transaction)
+	assert.NoError(t, err)
+
+	//Sign with the old supply key
+	_, err = tx.Execute(env.Client)
+	assert.ErrorContains(t, err, "exceptional precheck status INVALID_SUPPLY_KEY")
+
+}
+
+func createTokenWithKeysAndUpdateTokenKeyHelper(t *testing.T, createKeyType KeyType, updateKeyType KeyType, client *Client, newKey Key, initialKey Key, signerKey PrivateKey, verificationMode TokenKeyValidation) (TransactionResponse, TokenID, error) {
+	// Create Fungible token with keys
+	tx := NewTokenCreateTransaction().
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetTreasuryAccountID(client.GetOperatorAccountID()).
+		SetFreezeDefault(false)
+
+	switch createKeyType {
+	case WIPE_KEY:
+		tx.SetWipeKey(initialKey)
+	case KYC_KEY:
+		tx.SetKycKey(initialKey)
+	case SUPPLY_KEY:
+		tx.SetSupplyKey(initialKey)
+	case FREEZE_KEY:
+		tx.SetFreezeKey(initialKey)
+	case FEE_SCHEDULE_KEY:
+		tx.SetFeeScheduleKey(initialKey)
+	case PAUSE_KEY:
+		tx.SetPauseKey(initialKey)
+	case METADATA_KEY:
+		tx.SetMetadataKey(initialKey)
+	case ADMIN_KEY:
+		tx.SetAdminKey(initialKey)
+	case LOWER_PRIVILEGE:
+		tx.SetWipeKey(initialKey).
+			SetKycKey(initialKey).
+			SetSupplyKey(initialKey).
+			SetFreezeKey(initialKey).
+			SetFeeScheduleKey(initialKey).
+			SetPauseKey(initialKey).
+			SetMetadataKey(initialKey)
+	case ALL:
+		tx.SetWipeKey(initialKey).
+			SetKycKey(initialKey).
+			SetSupplyKey(initialKey).
+			SetFreezeKey(initialKey).
+			SetFeeScheduleKey(initialKey).
+			SetPauseKey(initialKey).
+			SetMetadataKey(initialKey).
+			SetAdminKey(initialKey)
+	case ALL_DIFFERENT_ADMIN_KEY:
+		tx.SetWipeKey(initialKey).
+			SetKycKey(initialKey).
+			SetSupplyKey(initialKey).
+			SetFreezeKey(initialKey).
+			SetFeeScheduleKey(initialKey).
+			SetPauseKey(initialKey).
+			SetMetadataKey(initialKey).
+			SetAdminKey(signerKey)
+	}
+	frozenTx, err := tx.FreezeWith(client)
+	require.NoError(t, err)
+	resp, err := frozenTx.Sign(signerKey).Execute(client)
+	require.NoError(t, err)
+	receipt, err := resp.SetValidateStatus(true).GetReceipt(client)
+	require.NoError(t, err)
+
+	resp, err = updateTokenKeysHelper(t, *receipt.TokenID, updateKeyType, client, newKey, signerKey, verificationMode)
+	return resp, *receipt.TokenID, err
+}
+
+func updateTokenKeysHelper(t *testing.T, tokenID TokenID, updateKeyType KeyType, client *Client, newKey Key, signerKey PrivateKey, verificationMode TokenKeyValidation) (TransactionResponse, error) {
+	privateKey, _ := newKey.(PrivateKey)
+	// Update the key
+	tx2 := NewTokenUpdateTransaction().
+		SetTokenID(tokenID).
+		SetKeyVerificationMode(verificationMode)
+
+	switch updateKeyType {
```

Comment:
Maybe the switch can be reused in a function taking in a transaction and a the KeyType

Reply:
Also the NONE value is never checked against in the switch